conversion/resource_generation.go

```diff
@@ -0,0 +1,413 @@
+		// Destination connection profiles do not need to be validated as for their creation gcs bucket will also be created
+		if profile.DstConnectionProfile.Name != "" && !validateOnly {
+			// Check if destination connection profile exists
+			dstProfileExists, err = s.connectionProfileExists(ctx, projectId, profile.DstConnectionProfile.Name, profile.DstConnectionProfile.Location, connectionProfiles, dsClient, &s)
+			if err != nil {
+				return sourceProfilesToCreate, dstProfilesToCreate, err
+			}
+		}
+		if !sourceProfileExists {
+			id := profile.SrcConnectionProfile.Name
+			if id == "" {
+				id = "CN-" + uuid.New().String()
+				profile.SrcConnectionProfile.Name = id
+			}
+			if profile.SrcConnectionProfile.Location == "" {
+				profile.SrcConnectionProfile.Location = region
+			}
+			sourceProfilesToCreate = append(sourceProfilesToCreate, &ConnectionProfileReq{
+				ConnectionProfile: ConnectionProfile{
+					ProjectId:    projectId,
+					DatashardId:  profile.DataShardId,
+					Id:           profile.SrcConnectionProfile.Name,
+					IsSource:     true,
+					Host:         profile.SrcConnectionProfile.Host,
+					Port:         profile.SrcConnectionProfile.Port,
+					Password:     profile.SrcConnectionProfile.Password,
+					User:         profile.SrcConnectionProfile.User,
+					Region:       profile.SrcConnectionProfile.Location,
+					ValidateOnly: validateOnly},
+				ctx: ctx,
+			})
+		}
+		if !dstProfileExists && !validateOnly {
+			id := profile.DstConnectionProfile.Name
+			if id == "" {
+				id = "CN-" + uuid.New().String()
+				profile.DstConnectionProfile.Name = id
+			}
+			if profile.DstConnectionProfile.Location == "" {
+				profile.DstConnectionProfile.Location = region
+			}
+			dstProfilesToCreate = append(dstProfilesToCreate, &ConnectionProfileReq{
+				ConnectionProfile: ConnectionProfile{
+					ProjectId:    projectId,
+					DatashardId:  profile.DataShardId,
+					Id:           id,
+					IsSource:     false,
+					Region:       profile.DstConnectionProfile.Location,
+					BucketName:   "GCS-" + uuid.New().String(),
+					ValidateOnly: false},
+				ctx: ctx,
+			})
+		}
+	}
+	return sourceProfilesToCreate, dstProfilesToCreate, err
+}
+
+// 1. For each datashard, check if source and destination connection profile exists or not
+// 2. If source connection profile doesn't exists create it or validate if creation is possible.
+// 3. If validation is false and destination connection profile doesn't exists create a corresponding gcs bucket and then a destination connection profile
+func (r ResourceGenerationStruct) CreateResourcesForShardedMigration(ctx context.Context, projectId string, instanceName string, validateOnly bool, region string, sourceProfile profiles.SourceProfile, dsClient *datastream.Client) error {
+	var sourceProfilesToCreate []*ConnectionProfileReq
+	var dstProfilesToCreate []*ConnectionProfileReq
+
+	// Fetches list with resources which do not exist and need to be created
+	s := ResourceGenerationStruct{}
```

Comment:
`s` is not readable. Please rename appropriately, applicable to other places

Reply:
changed in new PR